API/common/service_impl/learn_knowledge_index.py

```diff
@@ -142,14 +226,56 @@ def merge_chunks_by_command(chunks_list):
             for param in new_required_parameters:
                 if param not in existing_required_parameters:
                     existing_required_parameters.append(param)
+
+            existing_chunk['score'] = max(existing_chunk['score'], chunk['score'])
         else:
-            chunk.pop('score', None)
             summary_dict[command] = chunk
 
     merged_chunks = list(summary_dict.values())
     return merged_chunks
 
 
+def _pop_param_in_chunk(param, chunk):
+    for param_group in ['required parameters', 'optional parameters']:
+        for idx, chunk_param in enumerate(chunk.get(param_group, [])):
+            for chunk_param_option in chunk_param['name'].split():
+                if param == chunk_param_option:
+                    return chunk[param_group].pop(idx)
+    return None
+
+
+def _find_top_n_similar_params(param, chunk_param_group, top_n=3):
+    params = []
+    for chunk_param in chunk_param_group:
+        chunk_param_copy = chunk_param.copy()
+        scores = []
+        for chunk_option in chunk_param['name'].split():
+            score = fuzz.token_sort_ratio(chunk_option, param) / 100
+            scores.append(score)
+        if scores and max(scores) >= keyword_similarity_score:
+            chunk_param_copy['score'] = max(scores)
+            params.append(chunk_param_copy)
+    return sorted(params, key=lambda p: p['score'], reverse=True)[:top_n]
```

Comment:
Could we ensure that the similarity score of the top N items is greater than `keyword_similarity_score`?